pandas/tests/test_strings.py

```diff
@@ -2760,6 +2761,17 @@ def test_str_cat_raises_intuitive_error(self):
         with tm.assert_raises_regex(ValueError, message):
             s.str.cat('    ')
 
+    def test_str_cat_align(self):
+        # https://github.com/pandas-dev/pandas/issues/18657
```

Comment:
needs another case where this would produce nans on some elements (e.g. the original issue)

Reply:
Will add